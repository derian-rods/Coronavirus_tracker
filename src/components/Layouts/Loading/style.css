.loading__screend{
    width: 100vw;
    height: 100vh;
    display: flex;
    justify-content: center;
    align-items: center;
}
.loading__container{
    width: 24rem;
    padding: 4em;
}
.loading__img{
    width: 100%;
    animation: infinite-spinning 4s linear infinite;
}

@keyframes infinite-spinning{
    from{
        transform: rotate(0deg);
    }to{
        transform: rotate(360deg);    
    }
}