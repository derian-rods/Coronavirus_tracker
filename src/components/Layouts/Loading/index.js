import React from 'react'
import './style.css'
export const LoadingScreen = () => {
    return (
        <div className="loading__screend">
            <div className="loading__container">
                <img  className="loading__img" src='https://www.flaticon.com/svg/static/icons/svg/2760/2760528.svg' alt="corona virus tracker"/>
            </div>
        </div>
    )
}
