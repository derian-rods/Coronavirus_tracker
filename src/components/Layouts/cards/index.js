import React from 'react';
import PropTypes from 'prop-types';
import './style.css';
export const Card = ({title, total, increment, icon, icon2, color}) => {
    return (
        <div className="card">
           <div className="card__body">
           <h4 className="card__title">{title}</h4>
            <h3 className={color}>
              {
                  increment === 0 ? (
                    <i className="far fa-smile"></i>
                  ):(

                  <span>+{increment && new Intl.NumberFormat("es-ES").format(increment)} <i className={icon2}></i></span>
                  )
              }
            </h3>
            <span className="card__text">
                {new Intl.NumberFormat("es-ES").format(total)} total <i className={icon}></i>
            </span> <br/>
           </div>  
        </div>
    )
}
Card.propType = {
  title: PropTypes.string.isRequired,
  total: PropTypes.number,
  increment: PropTypes.number,
  icon: PropTypes.string,
  icon2: PropTypes.string,
  color: PropTypes.string
}