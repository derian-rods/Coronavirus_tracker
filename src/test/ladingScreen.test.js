import React from 'react';
import '@testing-library/jest-dom';
import {LoadingScreen} from '../components/Layouts/Loading';
import {shallow} from 'enzyme';

describe('test for LoadingScreen component', () => {

    test('should show <LoadingScreen /> correctly ', () => {
        const wrapper = shallow(<LoadingScreen />);
        expect(wrapper).toMatchSnapshot();
    });
})
