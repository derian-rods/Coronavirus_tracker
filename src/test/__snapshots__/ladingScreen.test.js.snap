// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for LoadingScreen component should show <LoadingScreen /> correctly  1`] = `
<div
  className="loading__screend"
>
  <div
    className="loading__container"
  >
    <img
      alt="corona virus tracker"
      className="loading__img"
      src="virus.svg"
    />
  </div>
</div>
`;
